### ISSUE TITLE
Convert to switch replaces with broken code.

### ISSUE THREAD

**Issue by JohanLarsson** [dotnet/roslyn#37035]
```cs
if (o is string text &&
    int.TryParse(text, out var n) &&
    n < 5 && n > -5)
{
}
else
{
}
```

Is refactored to:

```cs
switch (o)
{
    case string text when int.TryParse(text, out var n) &&
    n < 5:
        break;
    default:
        break;
}
```

Not ideal that it trims off `&& n > -5`

I expected:

```cs
switch (o)
{
    case string text when int.TryParse(text, out var n) &&
                        n < 5 && n > -5:
        break;
    default:
        break;
}
```

![Animation](https://user-images.githubusercontent.com/1640096/60790007-bf4df380-a160-11e9-92a9-2f1621060833.gif)

**Comment by CyrusNajmabadi** [comment]
I think the purpose is to speedily get you into switch form.  You may have to do some fixups as necessary if your existing constructs use features (like larger scoping) that aren't well supported by a switch expression.

**Comment by jnm2** [comment]
I agree, getting 90% of the way there is better than being forced to do all the typing myself.

Looks like best effort could be significantly improved though. There's no logical reason IMO that `&&` should be disappearing. It might even be possible to make the refactoring understand the variable scoping change.

**Comment by jnm2** [comment]
I missed that a whole operand was gone after the `&&`, not just the operator. That's a more serious bug because it could be missed if you were moving fast and didn't have a test covering it.

**Comment by CyrusNajmabadi** [comment]
Note: the bit about the clause being stripped off should def just be fixed.  I wasn't referring to that bit :)

**Comment by alrz** [comment]
this should be fixed by #38238

**Comment by CyrusNajmabadi** [comment]
Yup.  Have validated this is fixed.  LMK if there are other issues we need to closeout.  Thanks!
